test/priority_queue_test.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <openssl/opensslconf.h>
+#include <internal/priority_queue.h>
+#include <openssl/err.h>
+#include <openssl/crypto.h>
+
+#include "internal/nelem.h"
+#include "testutil.h"
+
+#define MAX_SAMPLES 500000
+
+DEFINE_PRIORITY_QUEUE_OF(size_t);
+
+static size_t num_rec_freed;
+
+static int size_t_compare(const size_t *a, const size_t *b)
+{
+    if (*a < *b)
+        return -1;
+    if (*a > *b)
+        return 1;
+    return 0;
+}
+
+static int qsort_size_t_compare(const void *a, const void *b)
+{
+    return size_t_compare((size_t *)a, (size_t *)b);
+}
+
+static int qsort_size_t_compare_rev(const void *a, const void *b)
+{
+    return size_t_compare((size_t *)b, (size_t *)a);
+}
+
+static void free_checker(ossl_unused size_t *p)
+{
+    num_rec_freed++;
+}
+
+static int test_size_t_priority_queue_int(int reserve, int order, int count,
+                                          int remove, int random, int popfree)
+{
+    PRIORITY_QUEUE_OF(size_t) *pq = NULL;
+    static size_t values[MAX_SAMPLES], sorted[MAX_SAMPLES], ref[MAX_SAMPLES];
+    size_t n;
+    int i, res = 0;
+    static const char *orders[3] = { "unordered", "ascending", "descending"};
```

Comment:
nit: space after opening brace, so should be one before closing brace (or remove it)

Reply:
done.